alarm_trap_sender.cpp

```diff
@@ -138,8 +142,40 @@ unsigned long AlarmFilter::current_time_ms()
   return ts.tv_sec * 1000 + (ts.tv_nsec / 1000000);
 }
 
+alarmActiveTable_SNMPDateTime* AlarmTrapSender::alarm_time_issued(void)
+{
+  alarmActiveTable_SNMPDateTime* d_time = new alarmActiveTable_SNMPDateTime();
+  struct timespec ts;
+  struct tm *timing;
+  time_t rawtime;
+  char plus_or_minus;
+  int daylight_savings;
+
+  clock_gettime(CLOCK_REALTIME, &ts);
+  rawtime = ts.tv_sec;
+  timing = gmtime(&rawtime);
+  // This updates the variable timezone declared within time.h which
+  // tells us how offset (in seconds) from UTC we are.
+  localtime(&rawtime);
+  plus_or_minus = (timezone > 0) ? '+' : '-';
+  daylight_savings = abs(timezone) / 3600;
```

Comment:
This isn't the right name for this variable - it's the offset from UTC, not the flag indicating whether daylight savings is in effect. (For example, I think this will always be 6 in Bangladesh, which doesn't have daylight savings time.)